[Jerry Ho] Duke Increments

itsjerryho opened the pull request:
Include Commits for Week 2

itsjerryho committed:
Add Scanner and Implement Greet, Echo, Exit

itsjerryho committed:
Import ArrayList

itsjerryho committed:
Add ability to store text and display back to user when requested

itsjerryho committed:
Add Task Class

itsjerryho committed:
Add markAsDone to Task Class

itsjerryho committed:
Add getDescription to Task Class

itsjerryho committed:
Change ArrayList from String to Task and modify display of list items

itsjerryho committed:
Create method printTask and add mark as done feature

itsjerryho committed:
Remove printTask from Duke.java and Override Task's toString method

itsjerryho committed:
Add Deadline class

itsjerryho committed:
Add ToDo Class

itsjerryho committed:
Add Event Class

itsjerryho committed:
Integrate Deadline, ToDo and Event to Duke

itsjerryho committed:
Modify EXPECTED.txt, input.txt to include test case.

Edit runtest.txt to javac all java files and replace code with symbols in Task.java

itsjerryho committed:
Add DukeException.java

itsjerryho committed:
Seperate mainLogic from main() and create a createTask method. Add basic Exceptionhandling

itsjerryho committed:
Add delete function

itsjerryho committed:
Add additional exception handling, fixed early exiting bug after exception handling

itsjerryho committed:
Add printMessage method to output messages with lines

itsjerryho committed:
Add toText method to Deadline, Event and Task classes

itsjerryho committed:
Add writeToFile, saveToDisk and readFileContents method to Duke.java

- add various imports

itsjerryho committed:
Add gitignore for data.txt

itsjerryho committed:
Add new constructor method to Deadline, Task, Event and ToDo to support boolean isDone

itsjerryho committed:
Add saveDataToList method to Duke.java

- saveDataToList is able to add tasks (in String form) to ArrayList

itsjerryho committed:
Add LocalDate support to Deadline.java and Duke.java

- Modify Deadline.java to store dates as a LocalDate object and time as a String.
- Update Deadline handling in Duke.java

itsjerryho committed:
Merge branch 'branch-Level-7'

itsjerryho committed:
Merge branch 'branch-Level-8'

itsjerryho committed:
Modify Deadline.java to make it compatible with DateTime implementation and fix infinite loop bug.

- Fix bug that is preventing bye command from exiting the program

lerxcl commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,201 @@
+    }
+
+    public static void printMessage(String s) {
+        String output = String.format("____________________________________________________________\n%s\n" +
```
The "+" should be in the next line instead.

lerxcl commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,201 @@
+    public static void saveToDisk(ArrayList<Task> lst) {
+        String filePath = "./data/data.txt";
+        String info = "";
+        for (int i = 0; i < lst.size(); i ++) {
```
There should not be a spacing between "i" and "++".

lerxcl commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,201 @@
+import java.util.ArrayList;
+import java.io.File;
+
 public class Duke {
```
I have noticed that your Duke class may have too many methods (a little cluttered), and seeing that you have not merged "branch-A-OOP" into master, I am not sure whether you have create new classes to adhere to OOP principles?

lerxcl commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,201 @@
+        String greetings = "Hello! I'm Duke, your personal assistant.\nWhat can I do for you?";
+        printMessage(greetings);
+
+        ArrayList<Task> lst = new ArrayList<>();
```
A better practice might be to split the instantiation of "lst".

After your import statements, it could be List<Task> lst;
Then in your Duke constructor: this.lst = new ArrayList<>();

lerxcl commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,201 @@
+    }
+    public static void mainLogic(Scanner sc, ArrayList<Task> lst) {
+
+        while (true) {
```
It might not be too good of practice to use "while (true)", as it is hard to tell what is your terminating condition (and there is a chance of an infinite loop)?

lerxcl commented on src/main/java/DukeException.java:
```diff
@@ -0,0 +1,5 @@
+public class DukeException extends Exception{
```
A spacing after "Exception" will look nicer!

lerxcl commented on src/main/java/ToDo.java:
```diff
@@ -0,0 +1,14 @@
+public class ToDo extends Task{
```
Same as mentioned in DukeException class.

lerxcl commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,40 @@
+    }
+
+    public String toText(String type) {
+        int doneInt = this.isDone ? 1 : 0;
```
It might be better to use Enum to represent 1 and 0, as it might look like "magic numbers".

lerxcl reviewed (commented):
LGTM. Overall, the code logic flows well. However, you should do more OOP and correct some of the syntax, good effort!

hyngkng commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,201 @@
+        }
+        return newTask;
+    }
+    public static void mainLogic(Scanner sc, ArrayList<Task> lst) {
```
Perhaps the method name could have been a verb instead? Eg. parseCommand.

michael-setia replied:
Yes, I also found this in other classes i.e. method invalidInput() in Duke class perhaps can also be changed to a verb?

hyngkng reviewed (commented):
I like your overall naming of methods, just have to ensure all of them are consistent with the coding standard! :)

michael-setia commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,201 @@
+
+    public static void printMessage(String s) {
+        String output = String.format("____________________________________________________________\n%s\n" +
+                "____________________________________________________________\n",s);
```
I think there should be a space after the comma?

michael-setia commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,40 @@
+
+    public String toText(String type) {
+        int doneInt = this.isDone ? 1 : 0;
+        return String.format("%s | %d | %s", type, doneInt, this.description);
```
I like how you use String.format() here. I think it looks cleaner than using "+" for the strings concatenation 😄

michael-setia commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,25 @@
+public class Event extends Task {
+    protected String at;
```
Maybe the attribute name "at" can be changed to a noun like "date" or "eventDate" to enhance readability?

michael-setia commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,201 @@
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.File;
```
I think the import statements can be grouped based on their package and add one blank line between the groups to enhance readability?

michael-setia reviewed (commented):
Overall, I think the coding standards are mostly adhered except for some of the missing spaces and naming. Keep up the good work! 😄

itsjerryho committed:
Create Ui.java

itsjerryho committed:
Create TaskList.java

itsjerryho committed:
Create Storage.java

itsjerryho committed:
Create Parser.java

itsjerryho committed:
Create Command.java

itsjerryho committed:
Create AdditionalInfo.java

- Stores additional info during Parsing

itsjerryho committed:
Minor edits after reading Comments on GitHub

- Introduce constant variables to prevent magic numbers
- Fix the spacing to adhere to coding standards

itsjerryho committed:
Restructure Duke by using OOP principles

itsjerryho committed:
Add all classes to Duke package except Duke.java

itsjerryho committed:
Create Common.java to store commonly used constants

itsjerryho committed:
Add jUnit test for Parser.java

- Include basic testing without exceptions

itsjerryho committed:
Add method to create folder and/or data file in the event that either one is missing during initialization

itsjerryho committed:
Fix bug in the case of invalid commands

itsjerryho committed:
Add javadocs to some classes

- Modify access to some methods in Duke.java

itsjerryho committed:
Modify files to adhere to Java coding standards.

itsjerryho committed:
Add find command to Duke

itsjerryho committed:
Merge branch 'branch-A-JavaDoc'

itsjerryho committed:
Merge branch 'branch-A-CodingStandard'

itsjerryho committed:
Merge branch 'branch-Level-9'

itsjerryho committed:
Add Gradle support

itsjerryho committed:
Merge branch 'branch-A-Gradle'

itsjerryho committed:
Add support for Gradle Checkstyle

itsjerryho committed:
Merge branch 'branch-A-CheckStyle'

itsjerryho committed:
Add DialogBox, Launcher, Main and MainWindow for GUI support

itsjerryho committed:
Add images and fxml files for DialogBox and MainWindow

itsjerryho committed:
Add support for Duke GUI

- Command Line Interfaace is not working at the moment

itsjerryho committed:
Make minor edits

- Remove scanner from Ui.java
- Add a few Todos

itsjerryho committed:
Fix bug in Duke.java

- Program previously exit without saving task to data.txt

itsjerryho committed:
Make minor changes to build.gradle and fxml files to remove warnings

itsjerryho committed:
Add assertions to TaskList.java

itsjerryho committed:
Add assertion to AdditionalInfo.java

itsjerryho committed:
Add assertion to build.gradle

itsjerryho committed:
Create TimedTask

- Edit Deadline and Event to extend TimedTask

itsjerryho committed:
Refactor Code

- Remove AdditionalInfo and Common classes
- Add Specific Command Classes (Add, Delete, Done, Exit, Find, Invalid, List, PrintTask)

itsjerryho committed:
Refactor Code

- Shift execute command logic out of Duke.java into the Command class itself
- Remove additionalInfo from Command class and add storage, taskList and userInput to Command class
- Create TimedTask, which is a super class for both Deadline and Event classes
- Shift main parsing logic out of Parser into specific Command classes
- Edit Storage class to handle change in Parsing style
- Create enum for Task.java
- Update Ui and remove Scanner and CLI related commands from Ui
- Remove AdditionalInfo from ParserTest (still in progress)

itsjerryho committed:
Add enums for Task class

itsjerryho committed:
Merge pull request #1 from printinghelloworld/branch-A-Assertions

Merge Branch A Assertions

itsjerryho committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/AdditionalInfo.java

itsjerryho committed:
Merge branch 'branch-A-CodeQuality'

itsjerryho committed:
Merge branch 'A-Enums'

itsjerryho committed:
Implement BetterSearch feature

- Find is now able to search using keyword that is not case-sensitive

itsjerryho committed:
Merge pull request #3 from printinghelloworld/branch-C-BetterSearch

Implement BetterSearch feature

itsjerryho committed:
Add functionality to delete multiple tasks at one go

itsjerryho committed:
Merge branch 'branch-C-MassOps'

itsjerryho committed:
Minor edits to FindCommand

itsjerryho committed:
Edit README.md

itsjerryho committed:
Merge branch 'branch-A-UserGuide'

itsjerryho committed:
Add Javadocs and Checkstyle

itsjerryho committed:
Increase JUnit testing

- Add JUnit tests in ParserTest
- Add TaskListTest

itsjerryho committed:
Improve GUI

- Add padding
- Change photo of Duke

itsjerryho committed:
Final changes to Duke

itsjerryho committed:
Merge branch 'branch-A-BetterGui'
